src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -262,6 +270,9 @@ internal AzureResourceInfrastructure() : base(default!) { }
     public static partial class AzureUserAssignedIdentityExtensions
     {
         public static ApplicationModel.IResourceBuilder<AzureUserAssignedIdentityResource> AddAzureUserAssignedIdentity(this IDistributedApplicationBuilder builder, string name) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<T> WithAzureUserAssignedIdentity<T>(this ApplicationModel.IResourceBuilder<T> builder, ApplicationModel.IResourceBuilder<AzureUserAssignedIdentityResource> identityResourceBuilder)
+            where T : ApplicationModel.IComputeResource { throw null; }
```

Comment:
Does this make sense on anything other than `IAzureComputeResource?`

Reply:
... or is this to deal with self-managed K8S clusters were you might still want this ability to associate an identity even though you probably need to configure aad-pod-identity/add-workload-identity yourself?